src/emqx_client.erl

```diff
@@ -151,6 +155,23 @@
 %%------------------------------------------------------------------------------
 %% API
 %%------------------------------------------------------------------------------
+-spec(def_response(client(), function() | binary()) -> ok).
+def_response(Responser, Response) ->
+    gen_statem:call(Responser, {def_response, Response}).
+
+sub_response_topic(Client, Shared, RequestQoS, Topic) ->
```

Comment:
needs a spec.
and perhaps there is no need for a `Shared` boolean flag, rather extend `Topic` from `binary()` to `Topic :: binary() | ?SHARED(ShareGroup :: binary(), Topic :: binary())`
where ?SHARED is `-define(SHARED(GROUP, TOPIC), {GROUP, TOPIC})`.

Reply:
Good idea.